### ISSUE TITLE
train_network.pyで --network_dimと--network_alphaを指定しないと重みがnoneになる問題

### ISSUE THREAD

**Issue by iKora128** [kohya-ss/sd-scripts#529]
train_network.pyで --network_dimと--network_alphaを指定しないと重みがnoneになる問題

2週間前まで学習できましたが、本日学習させるとすべての出力画像が真っ黒になることが分かりました。
念のため、付属のgen_img_diffusers.pyでも稼働させましたが同じ現象です。

→train_network.pyで --network_dimと--network_alphaを指定すると改善しました
defaultでは前者が4, 後者が1の記憶があります。とりあえず前者を64, 後者を32に明示的に指定したところ治りました。

**Comment by iKora128** [comment]
[追記]
 --network_dimと--network_alphaを[8,4]で指定するとやはり真っ黒になりました。
 そもそも非推奨でしたっけ？見落としてるかも知れませんが、ここらへんの設定をもう一度確認したいと思います。
 （前は低いrankでもできたいたのは勘違いだったかな）

**Comment by iKora128** [comment]
返信ありがとうございます。
設定以下にしております。
学習自体は走るのですが、出力されたモデルで推論すると真っ黒になる感じです。
（diffusersではエラーメッセージを吐かず、WebUIでは忘れましたが、重みがない？みたいな記述が合った気がします。）
また[dim, alpha] = [4,1], [8,4], [16,8]でやってみたら黒のままで、[64,32]だと動くようになってました。
自分の環境のせい・もしくは他の原因かもしれないので、もう少しちゃんと調べてみようかと思います。お手数かけます。

accelerate launch --num_cpu_threads_per_process 1 /sd-scripts/train_network.py \
    --pretrained_model_name_or_path={path} \
    --dataset_config= ~/config-LoRA.toml \
    --output_dir={out_path} \
    --output_name={name} \
    --prior_loss_weight=1.0 \
    --max_train_epochs=10 \
    --save_every_n_epochs=10 \
    --learning_rate=1e-4 \
    --optimizer_type=AdamW8bit \
    --xformers \
    --network_module=networks.lora \
    --mixed_precision=fp16 \
    --gradient_checkpointing \
    --save_model_as=safetensors \
    --network_dim=8 \
    --network_alpha=4 \

**Comment by iKora128** [comment]
[追記]
すみません、学習ベースのモデルと推論のモデルを変えると治りました。
どうやら手元のモデルが破損してる可能性のようです。学習スクリプトの問題ではなかったので、ご迷惑おかけしました。
不思議なのでは[dim, alpha]が[32,16]以上だと機能するところです笑
お手数かけました。ちょっと自分の方でもモデルの破損について調べてみます。（何も推論・学習ともに何もエラーメッセージ返さないので気づきませんでした。申し訳ありません。）

**Comment by iKora128** [comment]
閉じたあとにすみません。
ベースモデル変えても[dim, alpha]=[8,4], [32, 16]ではなぜか出力画像が真っ白になることが分かりました。
モデルに依存しているのか自分にはちょっと分かりませんでした。
他の組み合わせ[4,1], [8,1], [12, 4], [16, 8]ではいけました。

**Comment by iKora128** [comment]
ご丁寧にありがとうございます。
今調査中で、自分の環境依存なのかきちんと調べてからまた返信させていただきます。
（SD1.4 baseで学習したところ[4, 1]でも真っ白になるLoRAが作成されてしまったので、依存ライブラリを調査したり、環境を見直している途中です。scripts自体は問題なく動作するのですが、連続で学習を回してるのがいけないのかもしれません。また色々分かり次第ご返信させていただきます。お手数かけます。）

**Comment by iKora128** [comment]
すみません、こちらで検証したところ、CUDA updateすると治りました。
環境依存のようでした。お手数掛けました。
（なぜscriptは走るのに、白になったり黒になったりは結局解決できず仕舞いでしたが。）
